backend/src/main/java/com/google/backend/servlets/AlertConfigurationServlet.java

```diff
@@ -102,4 +112,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     return parameters;
   }
+
+  public final class Configuration {
+    private final String user;
+    private final String dimension;
+    private final String metric;
+    private final String relatedDimension;
+    private final String relatedMetric;
+
+    private Configuration(String user, String dimension, String metric, String relatedDimension, String relatedMetric) {
```

Comment:
ooc, why is the constructor marked as private?
you seem to use the constructor in line 55?

Reply:
+1, does the code still work with private constructor?

Reply:
I just changed the constructor to public

Reply:
I changed it to public.